apps/server/src/modules/h5p-editor/temporary-file-storage/temporary-file-storage.ts

```diff
@@ -105,39 +57,23 @@ export class TemporaryFileStorage implements ITemporaryFileStorage {
 		rangeEnd?: number | undefined
 	): Promise<Readable> {
 		this.checkFilename(filename);
-		this.checkFilename(user.id);
-		const file = await open(join(this.path, user.id, filename)); // file is auto closed
-		const stream = file.createReadStream({
-			start: rangeStart,
-			end: rangeEnd,
-		});
-		return stream;
+		const tempFile = await this.repo.findByPath(user.id, filename);
+		const path = join(user.id, filename);
+		if (rangeStart === undefined) {
+			if (rangeEnd === undefined) {
+				return (await this.s3Client.get(path)).data;
+			}
+			rangeStart = 0;
+		} else if (rangeEnd === undefined) {
+			rangeEnd = tempFile.size - 1;
+		}
+		return (await this.s3Client.get(path, `${rangeStart}-${rangeEnd}`)).data;
```

Comment:
```suggestion
		if (rangeStart === undefined) {
			rangeStart = 0;
		}
		if (rangeEnd === undefined) {
			rangeEnd = tempFile.size - 1;
		}
		const response = await this.s3Client.get(path, `${rangeStart}-${rangeEnd}`);
		
		return response.data;
```
I think this would work the same and be a lot easier to read.

Alternatively you could even set the default value of rangeStart to 0. Then you don't need the if-statement. Unfortunately this does not work for rangeEnd. Not sure if it's cleaner to set the default or to handle them alike...